Purchases/Error Handling/ErrorDetails.swift

```diff
@@ -19,5 +19,7 @@ class ErrorDetails: NSObject {
     static let finishableKey: NSError.UserInfoKey = "finishable"
     static let readableErrorCodeKey: NSError.UserInfoKey = "readable_error_code"
     static let generatedByKey: NSError.UserInfoKey = "generated_by"
+    static let extraContextKey: NSError.UserInfoKey = "extra_context"
+    static let attachedErrorKey: NSError.UserInfoKey = "attached_error"
```

Comment:
Isn't this the same as `NSUnderlyingErrorKey` that we use to attach the underlying error that originated the error?

Reply:
It's similar and definitely could be the same, but it's not as easy as just using the `NSUnderlyingErrorKey` key. I decided to attach an error using a new key instead of using `NSUnderlyingErrorKey` because there was at least a couple places where we'd have an error with an underlying error already, and it might mean we'd need to re-parents errors. So instead of reparenting errors, I left the current underlying error as is, and attached a more specific error to the parent error. The current purpose being better debugging for us. I think we could improve this by reparenting errors later.

Reply:
Going back through the code again, I think I can remove the `attachedErrorKey` and re-parent things in a non-complicated way.

Reply:
Let me know if you have any questions about how that works since I did that originally